src/toil/wdl/wdl_synthesis.py

```diff
@@ -706,10 +706,13 @@ def write_function_dockercall(self, job):
         stdout = apiDockerCall(self, 
                                image={docker_image}, 
                                working_dir=tempDir, 
-                               parameters=["/root/{job_task_reference}_script.sh"], 
+                               parameters=[os.path.join(tempDir, "{job_task_reference}_script.sh")], 
                                entrypoint="/bin/bash", 
-                               user={docker_user},
-                               volumes={{tempDir: {{"bind": "/root"}}}})
+                               user={docker_user}, 
+                               stderr=True, 
+                               volumes={{tempDir: {{"bind": tempDir}}}})
```

Comment:
Are you sure this works? I've had no success doing this in the past, but maybe my understanding is incomplete here.

I've never been able to use Docker to bind-mount a directory into a nonexistent directory in the container. It runs the container, but silently fails to actually perform the mount. Which does makes sense: there's nowhere for it to be mounted. But even if you later create the directory, you can never get the bind to actually occur (again, this makes sense).

Is this happening here? It looks to me like you may be bind mounting to an empty dir. `tempDir` seems to be set from `fileStore.getLocalTempDir()`, which (IIRC) is usually under the workDir, which by default gets set to something like `${TMPDIR}/toil-${machine-id}/randomstring`.  That directory surely doesn't exist in the container.

Reply:
I remember trying this in the past and having the same experience!  I saw someone try it recently though and thought it wouldn't work but it does now?  For example, if you want to create a mythical:
`/home/lifeisaboutfishtacos/Desktop/dragon`
folder with some files, and you run:
`docker run -it -v /home/lifeisaboutfishtacos/Desktop/dragon:/home/lifeisaboutfishtacos/Desktop/dragon ubuntu`
and run:
`ls /home/lifeisaboutfishtacos/Desktop/dragon`
You will see the contents of your folder.  It also runs the unittests fine, but yes, I was surprised too!  I wish I'd always been able to do this.

Reply:
Wow, I just tried it on my laptop and it works! I literally tried it just last week on my desktop and it didn't work, which is why I was so sure.

This is probably a feature that was released with a new version of Docker. I think my desktop is running a slightly old version, but my laptop is running the latest.

Well, this is good to know. It certainly simplifies things quite a bit.